packages/smooth_app/lib/pages/product/common/product_query_page.dart

```diff
@@ -339,4 +342,12 @@ class _ProductQueryPageState extends State<ProductQueryPage> {
           onPressed: () => Navigator.pop(context),
         ),
       );
+
+  Future<void> refreshlist() async {
+    final ProductListSupplier? refreshSupplier =
+        widget.productListSupplier.getRefreshSupplier();
+    setState(
+      () => _model = ProductQueryModel(refreshSupplier!),
+    );
+  }
```

Comment:
From what I know about the RefreshIndicator widget it's probably good to return something here